### ISSUE TITLE
Arity lint errors

### ISSUE THREAD

**Issue by beoliver** [clj-kondo/clj-kondo#1408]
**version**

clj-kondo v2021.09.25

**platform**

intel mac

**problem**

Many of the `clojure-core` type definitions (defined in `clj-kondo.impl.types.clojure.core`) do not cause lint errors when called with the wrong number of arguments

**repro**

```
$ clj-kondo --lint - <<< '(cons)'
linting took 10ms, errors: 0, warnings: 0
```

I dug a little bit using the following in `clj-kondo.types-test`

```clojure
(defn generate-expected-errors [type-information]
  (reduce-kv (fn [tests symb {:keys [arities]}]
               (if-not (:varargs arities)
                 (if-let [numbers (keys arities)]
                   (let [max-arity (inc (apply max numbers))
                         unsupported-arities (remove
                                              (fn [n] (get arities n))
                                              (range 0 (inc max-arity)))
                         failing-test-strings (map (fn [n]
                                                     (let [args (take n (repeat "1"))]
                                                       (format "(%s %s)" symb (clojure.string/join " " args))))
                                                   unsupported-arities)]
                     (assoc tests symb (zipmap unsupported-arities failing-test-strings)))
                   tests)
                 tests))
             {} type-information))

(defn find-expressions-that-do-not-cause-lint-warnings [type-information]
  (let [expected-errors (generate-expected-errors type-information)]
    (reduce-kv (fn [acc sym arity-map]
                 (let [result (into {} (map (fn [[n example]]
                                              (when-not
                                                  (seq
                                                   (lint!
                                                    example
                                                    {:linters {:type-mismatch {:level :error}}}))
                                                [n example]))
                                            arity-map))]
                   (if (empty? result)
                     acc
                     (assoc acc sym result))))
               {}
               expected-errors)))
```

The keys in the resulting map are functions that had no lint errors when called with the wrong number of args.

```clojure
> (find-expressions-that-do-not-cause-lint-warnings clojure-core)
{seq {0 "(seq )", 2 "(seq 1 1)"},
 butlast {0 "(butlast )", 2 "(butlast 1 1)"},
 first {0 "(first )", 2 "(first 1 1)"},
 seq? {0 "(seq? )", 2 "(seq? 1 1)"},
 last {0 "(last )", 2 "(last 1 1)"},
 rest {0 "(rest )", 2 "(rest 1 1)"},
 char? {0 "(char? )", 2 "(char? 1 1)"},
 cons {0 "(cons )", 1 "(cons 1)", 3 "(cons 1 1 1)"},
 map? {0 "(map? )", 2 "(map? 1 1)"},
 nfirst {0 "(nfirst )", 2 "(nfirst 1 1)"},
 meta {0 "(meta )", 2 "(meta 1 1)"},
 string? {0 "(string? )", 2 "(string? 1 1)"},
 second {0 "(second )", 2 "(second 1 1)"},
 ffirst {0 "(ffirst )", 2 "(ffirst 1 1)"},
 nnext {0 "(nnext )", 2 "(nnext 1 1)"},
 not-every? {0 "(not-every? )", 1 "(not-every? 1)", 3 "(not-every? 1 1 1)"},
 instance? {0 "(instance? )", 2 "(instance? 1 1)"},
 with-meta {0 "(with-meta )", 2 "(with-meta 1 1)"},
 next {0 "(next )", 2 "(next 1 1)"},
 not-any? {0 "(not-any? )", 1 "(not-any? 1)", 3 "(not-any? 1 1 1)"},
 fnext {0 "(fnext )", 2 "(fnext 1 1)"}}
```

It can also be run on the `clojure-set` definitions with a `(require [clj-kondo.impl.types.clojure.set :refer [clojure-set]])`
`(find-expressions-that-do-not-cause-lint-warnings clojure-set)`

**expected behavior**
I would expect arity errors to be consistent. For example linting `(flter)` shows an error.
```
clj-kondo --lint - <<< '(filter)'
<stdin>:1:1: error: clojure.core/filter is called with 0 args but expects 1 or 2
linting took 9ms, errors: 1, warnings: 0
```

Would be happy to look into creating a PR
